modules/EED_SquareOnsiteOAuth.module.php

```diff
@@ -721,37 +734,81 @@ public static function decryptString(string $text, bool $sandbox_mode): string
     }
 
 
+    /**
+     * Clean the array of data from sensitive information.
+     *
+     * @param array $data
+     * @return array
+     */
+    private static function cleanDataArray(array $data): array
+    {
+        $sensitive_data = [
+            'access_token',
+            'refresh_token',
+            'nonce',
+        ];
+        foreach ($data as $name => $value) {
+            // for multi. arrays
+            if (is_array($value)) {
+                foreach ($value as $key => $val) {
+                    if (in_array($key, $sensitive_data)) {
+                        unset($data[ $key ]);
+                    }
+                }
+            }
+            if (in_array($name, $sensitive_data)) {
+                unset($data[ $name ]);
+            }
+        }
+        return $data;
+    }
+
+
     /**
      * Log an error, return a json message and exit.
      *
-     * @param EE_Payment_Method|boolean $squarePm
-     * @param null                      $errMsg
-     * @param bool                      $doExit Should we echo json and exit
-     * @param bool                      $alert Tells frontend to show an alert or not
+     * @param EE_Payment_Method|boolean $square_pm
+     * @param string                    $err_msg
+     * @param array                     $data
+     * @param bool                      $do_exit Should we echo json and exit
+     * @param bool                      $oauth
+     * @param bool                      $alert  Tells frontend to show an alert or not
```

Comment:
names for booleans that represent an action are more readable if the name reflects/infers the actions,
especially if they are completely self explanatory to the point where they don't require a comment 
(within reason of course, we don't want variable names that are tooooo long)

so maybe something like:

```suggestion
     * @param bool                      $echo_json_and_exit // or $echo_and_exit, or $echo_json, or $return_json
     * @param bool                      $using_oauth // or $oauth_enabled, or $oauth_active, or ???
     * @param bool                      $show_alert  Tells frontend to show an alert or not
```